### ISSUE TITLE
Prompt for reporting unhelpful error messages and contributing to improvements

### ISSUE THREAD

**Issue by turbolent** [onflow/cadence#3711]
### Issue to be solved

The current error messages are sometimes not helpful enough and could be improved. See e.g. #294 and #2050 

### Suggested Solution

When reporting error messages, prompt the developer to report unhelpful error messages and to contribute improvements.

This could be limited to off-chain use-cases, i.e. in developer tooling.

For example:

```
Was this error unhelpful? Do you maybe even have a suggestion for how it could be improved?
Please report it in this issue: <LINK>
```

**Comment by bluesign** [comment]
can we make some section in `http://cadence-lang.org` for errors, and explain each error there maybe?

something like ( from original issue references ) https://typescript.tv/errors/#ts1006 

we can link to the page in error message, and people can improve the description there also with PR

**Comment by turbolent** [comment]
Yeah, definitely a good idea. Rust does this too, e.g. https://doc.rust-lang.org/error_codes/E0015.html
